src/namespaces/db/vector.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Literal } from "../../Cypher";
+import { CypherProcedure } from "../../procedures/CypherProcedure";
+import type { Expr } from "../../types";
+import { normalizeVariable } from "../../utils/normalize-variable";
+
+/** Returns all labels in the database
+ * @see [Neo4j Documentation](https://neo4j.com/docs/operations-manual/current/reference/procedures/#procedure_db_index_vector_queryNodes)
+ * @group Procedures
+ */
+export function queryNodes(
+    indexName: string | Literal<string>,
+    numberOfNearestNeighbours: number,
+    query: Expr
+): CypherProcedure<"node" | "score"> {
+    const procedureArgs = getVectorArguments(indexName, numberOfNearestNeighbours, query);
+
+    return new CypherProcedure("db.index.vector.queryNodes", procedureArgs);
+}
+
+/** Returns all labels in the database
+ * @see [Neo4j Documentation](https://neo4j.com/docs/operations-manual/current/reference/procedures/#procedure_db.index.vector.queryRelationships)
+ * @group Procedures
+ */
+export function queryRelationships(
+    indexName: string | Literal<string>,
```

Comment:
As I see this method also supports a `Literal` for the indexName.

This seems to be fine, but it may be worth to add a test case for passing a literal instead of a string

Reply:
Will do!